source/tyr/tyr/resources.py

```diff
@@ -1113,3 +1116,28 @@ def delete(self, name=None):
             raise
         return ({}, 204)
 
+class Data(flask_restful.Resource):
+    def post(self, instance_name):
+        instance = models.Instance.query_existing().filter_by(name=instance_name).first_or_404()
+        if instance is None:
+            return {'message': 'bad instance {}'.format(instance_name)}, 404
+
+        if not request.files:
+            return {'message': 'the Data file is missing'}, 400
+        content = request.files['file']
+        logger = logging.getLogger(__name__)
```

Comment:
You should use `get_instance_logger`, it will configure the logger so we can keep trace of the instance: using a the instance name as tag in syslog, or using a file named from the instance with file log storage